src/Microsoft.AspNet.OData.Shared/DeltaSetOfT.cs

```diff
@@ -0,0 +1,357 @@
+        public DeltaSet(IList<string> keys)           
+        {
+            _keys = keys;            
+            _clrType = typeof(TStructuralType);            
+        }
+
+    
+       /// <inheritdoc/>      
+        protected override void InsertItem(int index, IDeltaSetItem item)
+        {
+            Delta<TStructuralType> deltaItem = item as Delta<TStructuralType>;
+
+            if (deltaItem == null)
+            {
+                throw Error.Argument("item", SRResources.ChangedObjectTypeMismatch, item.GetType(), typeof(TStructuralType));
+            }
+
+            base.InsertItem(index, item);
+        }
+
+        /// <summary>
+        /// Handler for users Create, Get and Delete Methods
+        /// </summary>
+        internal PatchMethodHandler<TStructuralType> PatchHandler { get; set; }
+
+
+        /// <summary>
+        /// Patch for DeltaSet, a collection for Delta<typeparamref name="TStructuralType"/>
+        /// </summary>
+        /// <param name="originalCollection">Original collection of the Type which needs to be updated</param>
+        /// /// <returns>DeltaSet response</returns>
+        public DeltaSet<TStructuralType> Patch(ICollection<TStructuralType> originalCollection)
+        {
+            PatchHandler = new DefaultPatchHandler<TStructuralType>(originalCollection);
+
+            return CopyChangedValues();
+        }
+
+        /// <summary>
+        /// Patch for DeltaSet, a collection for Delta<typeparamref name="TStructuralType"/>
+        /// </summary>     
+        /// <returns>DeltaSet response</returns>
+        public DeltaSet<TStructuralType> Patch(IPatchMethodHandler patchHandler)
+        {
+            this.PatchHandler = patchHandler as PatchMethodHandler<TStructuralType>;
+            return CopyChangedValues();
+        }
+
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes")]
+        internal DeltaSet<TStructuralType> CopyChangedValues()
+        {
+            //Here we are getting the keys and using the keys to find the original object 
+            //to patch from the list of collection
+
+            DeltaSet<TStructuralType> deltaSet = CreateDetlaSet();
+
+            foreach (Delta<TStructuralType> changedObj in Items)
+            {
+                DataModificationOperationKind operation = DataModificationOperationKind.Update;
+
+                //Get filtered item based on keys
+                TStructuralType original = null;
+                string errorMessage = string.Empty;
+                string geterrorMessage = string.Empty;
+
+                Dictionary<string, object> keyValues = new Dictionary<string, object>();
+
+                foreach (string key in _keys)
+                {
+                    object value;
+                    changedObj.TryGetPropertyValue(key, out value);
+
+                    if (value != null)
```

Comment:
>value != null [](start = 24, length = 13)

what if value is null? Shouldn't this be an error, rather than silently ignoring?

Reply:
Assume this would have already thorwn error before itself, or in ODL

Reply:
Then maybe replace the check with an assert?

Reply:
We should verify, especially if the user can create/pass in a changed object that they have created manually.

In 4.0, does ODL automatically populate keys based on the @id; i.e., for deleted resources?